object_detection/object_detection.py

```diff
@@ -0,0 +1,436 @@
+                draw_annotation(annotated_frame, annotated_frame_with_overlay, label, f"{confidence_scores[cls][i]:.2f}", bbox[i], colorPalette, objects_settings[cls], cls)
+        else:
+            draw_annotation(annotated_frame, annotated_frame_with_overlay, label, f"{confidence_scores[label]:.2f}", bbox, colorPalette, objects_settings[cls], cls)
+    alpha = 0.35
+    result_image = cv2.addWeighted(annotated_frame_with_overlay, alpha, annotated_frame, 1 - alpha, 0)
+    return result_image
+
+def get_total_objects(bboxes):
+    total_objects = 0
+    for _, bbox in bboxes.items():
+        total_objects += len(bbox) if isinstance(bbox[0], list) else 1
+    return total_objects
+
+def detect_objects(camera_id, task_settings, input_frame, frame_shape, roi=None):
+    if roi is not None:
+        input_frame = input_frame[roi[1]:roi[3], roi[0]:roi[2]]
+    objects_settings = task_settings.get("objects_to_detect")
+    objects = list(objects_settings.keys())
+    #annotated_objects holds all members of objects where their object_settings[show_boxes] or object_settings[show_labels] is True
+    annotated_objects = [obj for obj in objects if objects_settings[obj].get("show_boxes", False) or objects_settings[obj].get("show_labels", False)]
+    blur_enabled_objects = [obj for obj in objects if objects_settings[obj].get("enable_blur", False)]
+    tracking_enabled_objects = [obj for obj in objects if objects_settings[obj].get("enable_tracking", False)]
+    
+    class_assignments = get_class_assignments(objects)
+    
+    enable_tracking = any(objects_settings[obj].get("enable_tracking", False) for obj in objects_settings)
+    #needs_blur is true if both objects has any supported_blur_classes and if for any of the supported_blur_classes in objects_settings, enable_blur is True
+    needs_blur = any(obj in blur_enabled_objects for obj in supported_blur_classes)
+    
+    model, id_tracker = get_model_and_tracker(camera_id, class_assignments, enable_tracking)    
+    if enable_tracking:
+        results = model.track(input_frame, persist=True, verbose=False)
+    else:
+        results = model(input_frame, verbose=False)
+
+    colorPalette = get_colors(camera_id, annotated_objects, input_frame, frame_shape)
+    
+    detections, bboxes, objects_detected, confidence_scores = get_detections(results, class_assignments, objects_settings)
+    
+    bboxes, confidence_scores = update_tracker(objects_settings, id_tracker, detections, bboxes, confidence_scores)
+    
+    blurred_frame = blur_objects(needs_blur, blur_enabled_objects, tracking_enabled_objects, objects_detected, bboxes, camera_id, input_frame) #if blur_faces is False, this function will return the input_frame as is
+    result_image = draw_annotations_and_overlay(blurred_frame, bboxes, colorPalette, objects_settings, confidence_scores)
+
+    return result_image, bboxes, list(objects_detected), get_total_objects(bboxes)
+
+if __name__ == '__main__':
```

Comment:
can you add a test code here and also for line crossing? Also I would recommend using a vscode extension to prettify python code. Its very difficult to read the code if there are no spacing between the lines.